### ISSUE TITLE
Problem with generic type subclass injection

### ISSUE THREAD

**Issue by Soflar** [ninject/Ninject#66]
In your file
FormatExtentions.cs
line 162:                 sb.Append(type.Name.Substring(0, type.Name.LastIndexOf('`')));

you are looking for a [grave accent]. (like this: ù, but on a space, I can't insert one because then github thinks I'm opening a code block) in generic type names, assuming it is there. The injection setup will crash if it's not. This is the case when using a type like SomeClass< T >.ISomeInterface, because the "name" property will return "SomeInterface" as value, and then the substring method crashes because it cannot find the `.

So basically, when injecting a subtype of a generic class, it will return true on querying "IsGeneric" and then return a type name without a [grave accent] in it, crashing the binding.

I hope this was put clearly enough. Thanks for providing this excellent tool for free!

PS: I'm using the MVC3 library to go along with this, I hope the MVC4 equivalent will be out soon! :)
